src/main/java/edu/project/components/WeatherIsland.java

```diff
@@ -15,18 +14,32 @@ public class WeatherIsland extends JPanel {
 	public WeatherIsland(Context context) {
 		putClientProperty(FlatClientProperties.STYLE,
 				"background: tint(@background,50%); border: 16,16,16,16,shade(@background,10%),,16");
-		setOpaque(false);
-
-		JLabel label = new JLabel("Hello World!");
-		add(label);
 	}
 
 	@Override
 	protected void paintComponent(Graphics graphics) {
+		super.paintComponent(graphics);
 		Graphics2D graphics2d = (Graphics2D) graphics;
 		graphics2d.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
-		graphics2d.setColor(getBackground());
-		graphics2d.fillRoundRect(0, 0, getWidth(), getHeight(), 16 + 12, 16 + 12);
-		super.paintComponent(graphics2d);
+
+		graphics2d.setColor(new Color(220, 220, 220, 180)); // Example color, adjust as needed
+
+		int arcWidth = 24;
+		int arcHeight = 24;
+
+		// Draw the top large rectangle (3-Hour Forecast)
+		graphics2d.fillRoundRect(20, 20, getWidth() - 40, 100, arcWidth, arcHeight);
+
+		// Draw the bottom large rectangle (5-Day Forecast)
+		graphics2d.fillRoundRect(20, 140, getWidth() - 40, 100, arcWidth, arcHeight);
+
+		// Draw the four squares (info boxes)
+		int squareSize = 60;
+		int spacing = 20;
+
+		graphics2d.fillRoundRect(20, 260, squareSize, squareSize, arcWidth, arcHeight);
+		graphics2d.fillRoundRect(100, 260, squareSize, squareSize, arcWidth, arcHeight);
+		graphics2d.fillRoundRect(180, 260, squareSize, squareSize, arcWidth, arcHeight);
+		graphics2d.fillRoundRect(260, 260, squareSize, squareSize, arcWidth, arcHeight);
```

Comment:
Ideally, the child components should be java swing components which can be `this.add(...)`-ed in the `WeatherIsland`. Drawing the child components directly with the graphics context will limit us from being able to use any other java swing components (e.g. JButton, JLabel, etc.)